expression/integration_test.go

```diff
@@ -7598,3 +7598,65 @@ func TestCastJSONTimeDuration(t *testing.T) {
 		fmt.Sprintf("6 %s %s 12:12:12 12:12:12 TIME", nowDate, nowDate),
 	))
 }
+
+func TestIssue35184(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists ft;")
+	tk.MustExec("create table ft (tint int, tdou double, tdec decimal(22,9),tchar char(44));")
+	tk.MustExec("insert into ft values(1234567890,123467890.1234,123467890.1234,'123467890.1234');")
+	tk.MustExec("insert into ft values(1234567890,123467890.123456789,123467890.123456789,'123467890.123456789');")
+
+	result := tk.MustQuery("SELECT FROM_UNIXTIME(tchar) from ft;")
+	unixTime1 := "1973-11-30 08:38:10.123400"
+	unixTime2 := "1973-11-30 08:38:10.123457"
+	result.Check(testkit.Rows(unixTime1, unixTime2))
+
+	tk.MustExec("drop table if exists ft;")
+	tk.MustExec("create table ft (tint int, tdou double, tdec decimal(22,9),tchar varchar(44));")
+	tk.MustExec("insert into ft values(1234567890,123467890.1234,123467890.1234,'123467890.1234');")
+	tk.MustExec("insert into ft values(1234567890,123467890.123456789,123467890.123456789,'123467890.123456789');")
+	result = tk.MustQuery("SELECT FROM_UNIXTIME(tchar) from ft;")
+	result.Check(testkit.Rows(unixTime1, unixTime2))
+
+	tk.MustExec("drop table if exists ft;")
+	tk.MustExec("create table ft (tint int, tdou double, tdec decimal(22,9),tchar blob);")
+	tk.MustExec("insert into ft values(1234567890,123467890.1234,123467890.1234,'123467890.1234');")
+	tk.MustExec("insert into ft values(1234567890,123467890.123456789,123467890.123456789,'123467890.123456789');")
+	result = tk.MustQuery("SELECT FROM_UNIXTIME(tchar) from ft;")
+	result.Check(testkit.Rows(unixTime1, unixTime2))
+
+	tk.MustExec("drop table if exists ft;")
+	tk.MustExec("create table ft (tint int, tdou double, tdec decimal(22,9),tchar tinyblob);")
+	tk.MustExec("insert into ft values(1234567890,123467890.1234,123467890.1234,'123467890.1234');")
+	tk.MustExec("insert into ft values(1234567890,123467890.123456789,123467890.123456789,'123467890.123456789');")
+	result = tk.MustQuery("SELECT FROM_UNIXTIME(tchar) from ft;")
+	result.Check(testkit.Rows(unixTime1, unixTime2))
+
+	tk.MustExec("drop table if exists ft;")
+	tk.MustExec("create table ft (tint int, tdou double, tdec decimal(22,9),tchar mediumblob);")
+	tk.MustExec("insert into ft values(1234567890,123467890.1234,123467890.1234,'123467890.1234');")
+	tk.MustExec("insert into ft values(1234567890,123467890.123456789,123467890.123456789,'123467890.123456789');")
+	result = tk.MustQuery("SELECT FROM_UNIXTIME(tchar) from ft;")
+	result.Check(testkit.Rows(unixTime1, unixTime2))
+
+	tk.MustExec("drop table if exists ft;")
+	tk.MustExec("create table ft (tint int, tdou double, tdec decimal(22,9),tchar longblob);")
```

Comment:
Could you remove the irrelevant columns and tail semicolon

Reply:
done